core/src/main/java/hudson/model/TimeZoneProperty.java

```diff
@@ -105,7 +105,7 @@ public static String forCurrentUser() {
         }
 
         TimeZone tz = TimeZone.getTimeZone(tzp.timeZoneName);
-        if (tz.getID() != tzp.timeZoneName) {
+        if (!tz.getID().equals(tzp.timeZoneName)) {
```

Comment:
That one might be a real defect in some JVM implementations. Should we changelog it?